connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -759,3 +764,97 @@ async function updateParentsField({
     parents,
   });
 }
+
+export async function microsoftDeletionActivity({
+  connectorId,
+  nodeIdsToDelete,
+}: {
+  connectorId: ModelId;
+  nodeIdsToDelete: string[];
+}): Promise<Result<void, Error>> {
+  const connector = await ConnectorResource.fetchById(connectorId);
+  if (!connector) {
+    return new Err(new Error(`Connector ${connectorId} not found`));
+  }
+
+  const dataSourceConfig = dataSourceConfigFromConnector(connector);
+
+  async function deleteNodeAndChildren(
+    nodeId: string
+  ): Promise<Result<void, Error>> {
+    const node = await MicrosoftNodeResource.fetchByInternalId(
+      connectorId,
+      nodeId
+    );
+
+    if (!node) {
+      logger.warn({ connectorId, nodeId }, "Node not found for deletion");
+      return new Ok(undefined);
+    }
+
+    const { nodeType } = typeAndPathFromInternalId(nodeId);
+
+    if (nodeType === "file") {
+      try {
+        await deleteFromDataSource(dataSourceConfig, node.internalId);
+      } catch (error) {
+        logger.error(
+          { connectorId, nodeId, error },
+          `Failed to delete document ${node.internalId} from core data source`
+        );
+        return new Err(
+          new Error(`Failed to delete document ${node.internalId}`)
+        );
+      }
+    } else if (nodeType === "folder" || nodeType === "drive") {
+      const children = await node.fetchChildren();
+      for (const child of children) {
```

Comment:
would use concurrentExecutor there, otherwise might be too slow
Assume we'll some day delete 100K files because a user will deselect a big drive

Reply:
actually, using concurrentExecutor recursively might have the opposite effect of blasting the api too much...
IMHO we should probably reproduce the traversing logic of syncFiles inherited from gdrive, that handles that nicely
complex topic, happy to discuss IRL